src/CashuWallet.ts

```diff
@@ -102,30 +119,34 @@ class CashuWallet {
 		const { token } = cleanToken(getDecodedToken(encodedToken));
 		const tokenEntries: Array<TokenEntry> = [];
 		const tokenEntriesWithError: Array<TokenEntry> = [];
-		const mintKeys = new Map<string, MintKeys>([[this.mint.mintUrl, this.keys]]);
+		let newKeys: MintKeys | undefined;
 		for (const tokenEntry of token) {
 			if (!tokenEntry?.proofs?.length) {
 				continue;
 			}
 			try {
-				const keys = mintKeys.get(tokenEntry.mint) || (await CashuMint.getKeys(tokenEntry.mint));
-				if (!mintKeys.has(tokenEntry.mint)) {
-					mintKeys.set(tokenEntry.mint, keys);
-				}
-				const result = await this.receiveTokenEntry(tokenEntry, keys);
-				if (result?.proofsWithError?.length) {
+				const {
+					proofsWithError,
+					proofs,
+					newKeys: newKeysFromReceive
+				} = await this.receiveTokenEntry(tokenEntry);
+				if (proofsWithError?.length) {
 					tokenEntriesWithError.push(tokenEntry);
 					continue;
 				}
-				tokenEntries.push({ mint: tokenEntry.mint, proofs: [...result.proofs] });
+				tokenEntries.push({ mint: tokenEntry.mint, proofs: [...proofs] });
+				if (!newKeys) {
+					newKeys = newKeysFromReceive;
+				}
 			} catch (error) {
 				console.error(error);
 				tokenEntriesWithError.push(tokenEntry);
 			}
 		}
 		return {
 			token: { token: tokenEntries },
-			tokensWithErrors: tokenEntriesWithError.length ? { token: tokenEntriesWithError } : undefined
+			tokensWithErrors: tokenEntriesWithError.length ? { token: tokenEntriesWithError } : undefined,
+			newKeys
```

Comment:
shouldn't this be a map/array of MintKeys? Seems like we could get a weird case, where keys in the second token changed, and now we return it as the new keys of token 1

Reply:
If we return newKeys the currently active keyset changed .Token 1 could use an old keyset and token 2 could use a new keyset we now return the newKeys (used for token2) because newKeys are now the currently used keyset.

Reply:
what if the 2nd tokenentry is from a different mint

Reply:
we only return newKeys from the mint the wallet is using 
see https://github.com/cashubtc/cashu-ts/blob/874a372697077c29347fee6e6a704fb8ffea79da/src/CashuWallet.ts#L236

Reply:
I missed that! thanks for clarifying 🫡